src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java

```diff
@@ -99,26 +98,29 @@ public ResponseEntity<String> putUserCommonsByIdBuy(
 
         String body = mapper.writeValueAsString(userCommons);
         return ResponseEntity.ok().body(body);
-    }
+      }
 
-  @ApiOperation(value = "Sell a cow, totalWealth updated")
+  @ApiOperation(value = "Sell cows, totalWealth updated")
   @PreAuthorize("hasRole('ROLE_USER')")
   @PutMapping("/sell")
   public ResponseEntity<String> putUserCommonsByIdSell(
-          @ApiParam("commonsId") @RequestParam Long commonsId) throws NoCowsException, JsonProcessingException {
+    @ApiParam("commonsId") @RequestParam Long commonsId,
+    @ApiParam("numCows") @RequestParam int numCows) // Want to set 1 as the default when nothing is passed in
+      throws NoCowsException, JsonProcessingException {
         User u = getCurrentUser().getUser();
         Long userId = u.getId();
 
-        Commons commons = commonsRepository.findById(commonsId).orElseThrow( 
-          ()->new EntityNotFoundException(Commons.class, commonsId));
+        Commons commons = commonsRepository.findById(commonsId)
+          .orElseThrow(()->new EntityNotFoundException(Commons.class, commonsId));
         UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
-        .orElseThrow(
-            () -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
-
-
-        if(userCommons.getNumOfCows() >= 1 ){
-          userCommons.setTotalWealth(userCommons.getTotalWealth() + commons.getCowPrice());
-          userCommons.setNumOfCows(userCommons.getNumOfCows() - 1);
+          .orElseThrow(() -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
+
+        if(userCommons.getNumOfCows() >= 1){
+          //if (numCows > userCommons.getNumOfCows()) numCows = userCommons.getNumOfCows();
+          //numCows = (numCows < userCommons.getNumOfCows()) ? numCows : userCommons.getNumOfCows();
+          numCows = Math.min(numCows, userCommons.getNumOfCows());
```

Comment:
Consider also maxing this with 0, so that it can't be negative, and then making sure that the code doesn't need a special case for 0 (i.e. that nothing unexpected happens when numCows is zero).

Reply:
These changes highlighted above were made in PR #22, which I merged into my branch. The code for increasing total cows bought in `UserCommonsController.java` would change, so I was afraid of a merge conflict. What should I do in this case?

Reply:
I'm not sure what you are asking...

Let's set up a time to meet and discuss. Reach out to me on slack.